platform/source/mbed_os_timer.cpp

```diff
@@ -47,13 +47,12 @@ OsTimer *init_os_timer()
     // Locking not required as it will be first called during
     // OS init, or else we're a non-RTOS single-threaded setup.
     if (!os_timer) {
-#if MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER && DEVICE_USTICKER
-        os_timer = new (os_timer_data) OsTimer(get_us_ticker_data());
-#elif !MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER && DEVICE_LPTICKER
+#if DEVICE_LPTICKER && !MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER
         os_timer = new (os_timer_data) OsTimer(get_lp_ticker_data());
+#elif DEVICE_USTICKER
+        os_timer = new (os_timer_data) OsTimer(get_us_ticker_data());
 #else
-        MBED_ASSERT("OS timer not available - check MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER" && false);
-        return NULL;
+        error("OS timer not available - check MBED_CONF_TARGET_TICKLESS_FROM_US_TICKER");
```

Comment:
I think this should probably be `MBED_ERROR` these days, as unwieldy as it is. I suppose it makes sense to not be an assert though - this path can't produce a functioning image in any way, and it will be clearer than just a null pointer exception.

Reply:
Advantage if this change is to see the error message in the console!

Reply:
looks like error is still valid, although would assume adding a new error would use `MBED_ERROR`